examples/token_refresh/client.ex

```diff
@@ -0,0 +1,44 @@
+defmodule MyApp.TokenRefreshClient do
+  @moduledoc """
+  A client which authenticates via token, and refreshes the token on 403.
+  """
+
+  use Slipstream
+
+  def start_link(config) do
+    Slipstream.start_link(__MODULE__, config, name: __MODULE__)
+  end
+
+  @impl Slipstream
+  def init(config) do
+    new_socket()
+    |> assign(:config, config)
+    |> assign_token()
+    |> connect(config_with_token())
```

Comment:
It might make sense to wrap `connect/2` in a helper that makes the token, so something like:

```elixir
defp connect_with_token(socket) do
  new_token = make_new_token()

  socket =
    update(socket, :config, fn config ->
      uri =
        config
        |> Keyword.get(:uri)
        |> URI.parse()
        |> Map.put(:query, "token=#{new_token}")
        |> URI.to_string()
  
      Keyword.put(config, :uri, uri)
    end)

  connect(socket, socket.assigns.config)
end
```

So the connection would look like

```elixir
new_socket()
|> assign(:config, config)
|> connect_with_token()
```

What do you think?

Reply:
Love it! That feels much cleaner. I've pushed up those changes and added a `README`. Let me know what you think. Once everything has your stamp of approval, I'll add tests! 

Thanks!

Reply:
Nice, this looks great!